src/backend/revoker/indexer/queryAttestation.test.ts

```diff
@@ -0,0 +1,314 @@
+/**
+ * @jest-environment node
+ */
+
+import { beforeEach, describe, expect, it, jest } from '@jest/globals';
+import { DidUri } from '@kiltprotocol/sdk-js';
+import got from 'got';
+
+import { configuration } from '../../utilities/configuration';
+
+import { FetchedData, QUERY_SIZE } from './queryFromIndexer';
+import {
+  buildAttestationQueries,
+  queryExpiredAttestations,
+  QueriedAttestation,
+  fromDate,
+  AttestationInfo,
+} from './queryAttestations';
+
+const frozenNow = new Date();
+
+beforeAll(() => {
+  // Fake only the Date
+  jest.useFakeTimers({
+    doNotFake: [
+      'hrtime',
+      'nextTick',
+      'performance',
+      'queueMicrotask',
+      'requestAnimationFrame',
+      'cancelAnimationFrame',
+      'requestIdleCallback',
+      'cancelIdleCallback',
+      'setImmediate',
+      'clearImmediate',
+      'setInterval',
+      'clearInterval',
+      'setTimeout',
+      'clearTimeout',
+    ],
+    now: frozenNow,
+  });
+});
+afterAll(() => {
+  jest.restoreAllMocks();
+  jest.useRealTimers();
+});
+
+jest.mock('../../utilities/configuration', () => ({
+  configuration: {
+    indexer: { graphqlEndpoint: 'https://dev-indexer.kilt.io/' },
+  },
+}));
+
+let postResponse: FetchedData;
+jest.mock('got', () => ({
+  post: jest.fn().mockReturnValue({
+    json: () => postResponse,
+  }),
+}));
+
+beforeEach(() => {
+  jest.mocked(got.post).mockClear();
+  configuration.indexer.graphqlEndpoint = 'https://dev-indexer.kilt.io/';
+});
+
+function mockAttestations(numberOfAttestations: number) {
+  const mockedAttestations: QueriedAttestation[] = [];
+
+  for (let index = numberOfAttestations; index > 0; index--) {
+    mockedAttestations.push({
+      id: index.toString(),
+      claimHash: `0x${index.toString(16)}`,
+      cTypeId: `kilt:ctype:0x${index.toString(16)}`,
+      issuerId: configuration.did as DidUri,
+      delegationID: null,
+      payer: 'PapaStaat',
+      valid: true,
+      creationBlock: {
+        id: index.toString(),
+        hash: `0x${index.toString(16)}`,
+        timeStamp: new Date(index, 3, 21).toISOString(),
+      },
+      revocationBlock: null,
+      removalBlock: null,
+    });
+  }
+
+  return mockedAttestations;
+}
+
+describe('The function that queries the old attestations issued by SocialKYC from the Indexer', () => {
+  describe('queryExpiredAttestations()', () => {
+    describe('on positive cases', () => {
+      it("On the App's start: should request attestations from 1970 until a year ago from the Indexer", async () => {
+        const timeZero = new Date(0);
+        const aYearAgo = new Date(frozenNow.valueOf());
+        aYearAgo.setFullYear(aYearAgo.getFullYear() - 1);
+
+        // check the start value of `fromDate`
+        expect(fromDate).toEqual(timeZero);
+
+        const count = Math.floor(QUERY_SIZE * 0.2);
+        postResponse = {
+          data: {
+            blocks: {
+              totalCount: count,
```

Comment:
```suggestion
        const totalCount = Math.floor(QUERY_SIZE * 0.2);
        postResponse = {
          data: {
            blocks: {
              totalCount,
```
If the constant `count` is used in a single place to initialize a value `totalCount`, would it make sense to rename it to match? I also see this in all the followup tests. Alternatively, if it is used only once, do we even need it?

Reply:
Ok, eliminated the variable when it's used only once.